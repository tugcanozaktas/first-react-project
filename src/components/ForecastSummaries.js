/* eslint-disable react/prop-types */
import React from "react";
import ForecastSummary from "./ForecastSummary";

// eslint-disable-next-line react/prop-types
function ForecastSummaries({ forecasts, onForecastSelect }) {
  return (
    <div className="forecast-summaries">
      {forecasts.map((forecast) => (
        <ForecastSummary
          key={forecast.date}
          date={forecast.date}
          description={forecast.description}
          icon={forecast.icon}
          onSelect={onForecastSelect}
          temperature={forecast.temperature}
        />
      ))}
    </div>
  );
}

export default ForecastSummaries;
