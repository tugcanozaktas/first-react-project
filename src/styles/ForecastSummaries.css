.forecast-summaries{
    display: flex;
    justify-content:space-between;
    gap: 5px
}