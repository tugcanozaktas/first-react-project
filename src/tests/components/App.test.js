import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("App", () => {
  const onSelect = () => {};
  const validProps = [
    {
      date: 1702490400000,
      temperature: { max: 5, min: 4 },
      wind: { speed: 3, direction: "n" },
      humidity: 77,
      description: "Clouds",
      icon: 803,
    },
    {
      date: 1702512000000,
      temperature: { max: 4, min: 3 },
      wind: { speed: 2, direction: "sse" },
      humidity: 81,
      description: "Clouds",
      icon: 803,
    },
    {
      date: 1702598400000,
      temperature: { max: 7, min: 6 },
      wind: { speed: 3, direction: "w" },
      humidity: 82,
      description: "Clouds",
      icon: 804,
    },
    {
      date: 1702684800000,
      temperature: { max: 8, min: 7 },
      wind: { speed: 3, direction: "ssw" },
      humidity: 97,
      description: "Clouds",
      icon: 804,
    },
    {
      date: 1702771200000,
      temperature: { max: 11, min: 10 },
      wind: { speed: 6, direction: "sw" },
      humidity: 85,
      description: "Clouds",
      icon: 804,
    },
  ];

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummaries forecasts={validProps} onForecastSelect={onSelect} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
