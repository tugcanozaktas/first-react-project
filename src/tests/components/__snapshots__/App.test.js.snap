// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-sumamry"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Wed 13th Dec
      </div>
      <div
        class="forecast-summary__icon"
      >
        <i
          class="wi wi-day-cloudy"
          data-testid="forecast-icon"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        5°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Clouds
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-sumamry"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu 14th Dec
      </div>
      <div
        class="forecast-summary__icon"
      >
        <i
          class="wi wi-day-cloudy"
          data-testid="forecast-icon"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        4°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Clouds
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-sumamry"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Fri 15th Dec
      </div>
      <div
        class="forecast-summary__icon"
      >
        <i
          class="wi wi-day-cloudy"
          data-testid="forecast-icon"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        7°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Clouds
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-sumamry"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Sat 16th Dec
      </div>
      <div
        class="forecast-summary__icon"
      >
        <i
          class="wi wi-day-cloudy"
          data-testid="forecast-icon"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        8°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Clouds
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-sumamry"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Sun 17th Dec
      </div>
      <div
        class="forecast-summary__icon"
      >
        <i
          class="wi wi-day-cloudy"
          data-testid="forecast-icon"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        11°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Clouds
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
